File: clojure/src/spoonacular_api/specs/get_ingredient_information_200_response.clj
(ns spoonacular-api.specs.get-ingredient-information-200-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.parse-ingredients-200-response-inner-estimated-cost :refer :all]
            [spoonacular-api.specs.get-ingredient-information-200-response-nutrition :refer :all]
            )
  (:import (java.io File)))


(def get-ingredient-information-200-response-data
  {
   (ds/req :id) int?
   (ds/req :original) string?
   (ds/req :originalName) string?
   (ds/req :name) string?
   (ds/req :nameClean) string?
   (ds/req :amount) float?
   (ds/req :unit) string?
   (ds/req :unitShort) string?
   (ds/req :unitLong) string?
   (ds/req :possibleUnits) (s/coll-of string?)
   (ds/req :estimatedCost) parse-ingredients-200-response-inner-estimated-cost-spec
   (ds/req :consistency) string?
   (ds/req :shoppingListUnits) (s/coll-of string?)
   (ds/req :aisle) string?
   (ds/req :image) string?
   (ds/req :meta) (s/coll-of any?)
   (ds/req :nutrition) get-ingredient-information-200-response-nutrition-spec
   (ds/req :categoryPath) (s/coll-of string?)
   })

(def get-ingredient-information-200-response-spec
  (ds/spec
    {:name ::get-ingredient-information-200-response
     :spec get-ingredient-information-200-response-data}))

File: scala/src/main/scala/com/spoonacular/client/model/GetIngredientInformation200Response.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import com.spoonacular.client.model.BigDecimal
import com.spoonacular.client.model.GetIngredientInformation200ResponseNutrition
import com.spoonacular.client.model.ParseIngredients200ResponseInnerEstimatedCost
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param original 
 * @param originalName 
 * @param name 
 * @param nameClean 
 * @param amount 
 * @param unit 
 * @param unitShort 
 * @param unitLong 
 * @param possibleUnits 
 * @param estimatedCost 
 * @param consistency 
 * @param shoppingListUnits 
 * @param aisle 
 * @param image 
 * @param meta 
 * @param nutrition 
 * @param categoryPath 
 */
case class GetIngredientInformation200Response(id: Int,
                original: String,
                originalName: String,
                name: String,
                nameClean: String,
                amount: BigDecimal,
                unit: String,
                unitShort: String,
                unitLong: String,
                possibleUnits: Seq[String],
                estimatedCost: ParseIngredients200ResponseInnerEstimatedCost,
                consistency: String,
                shoppingListUnits: Seq[String],
                aisle: String,
                image: String,
                meta: Seq[Object],
                nutrition: GetIngredientInformation200ResponseNutrition,
                categoryPath: Seq[String]
                )

object GetIngredientInformation200Response {
    /**
     * Creates the codec for converting GetIngredientInformation200Response from and to JSON.
     */
    implicit val decoder: Decoder[GetIngredientInformation200Response] = deriveDecoder
    implicit val encoder: ObjectEncoder[GetIngredientInformation200Response] = deriveEncoder
}
